### PR TITLE
fix: Stop query in SQL Lab with impala engine

### LINKED ISSUE #20950: Stop button for queries doesn't work in SQL Lab when using SQL Lab with impala engine
I am currently having a problem: Stop button for queries doesn't work in SQL Lab when I use SQL Lab with impala engine.

#### How to reproduce the bug

1.Run any query in SQL Lab with impala-thriftserver as backend processing engine.
2.Press STOP button on the UI.
3.Check the query engine, the query will still be running. You will still get the results on front end when the query finishes.

### Expected results

The query should be killed on the query processing engine.

### Actual results

Query keeps running on the processing engine. You will still get the results on the front end when the query finishes.

### Environment

browser type and version: Chrome/Firefox
superset version: latest
python version: 3.8

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

 [√] I have checked the superset logs for python stacktraces and included it here as text if there are any.
 [√] I have reproduced the issue with at least the latest released version of superset.
 [√] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by wanghong1314**
fix(sqllab): Stop button for queries doesn't work in SQL Lab when using SQL Lab with impala engine and adding Progress Information

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

BEFORE
image

AFTER
image

progress info
image

### ADDITIONAL INFORMATION

- [x] Has associated issue: fixes #20950
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by wanghong1314** [comment]
@eschutho @villebro Please review the code again. Thank you

File: superset/config.py [hunk 1]
```diff
@@ -1129,8 +1129,10 @@ def CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC(  # pylint: disable=invalid-name
 TRACKING_URL_TRANSFORMER = lambda url: url
 
 
-# Interval between consecutive polls when using Hive Engine
-HIVE_POLL_INTERVAL = int(timedelta(seconds=5).total_seconds())
+# customize the polling time of each engine. The default time is 5 seconds
+DB_POLL_INTERVAL_SECONDS = {
+    "hive": int(timedelta(seconds=5).total_seconds()),
+}
```
Comment (villebro) [inline superset/config.py:1135]: We can't really define defaults in `config.py`. Imagine if I want to override the poll interval for "impala" and add the following to `superset_config.py`:

```
DB_POLL_INTERVAL_SECONDS = {"impala": 1}
```

In this case the default for "hive" in `config.py` would be lost. This is why I recommended not defining the defaults in `config.py`, and rather placing them in their respective db engine specs.

File: docker/pythonpath_dev/superset_config.py [hunk 2]
```diff
@@ -106,6 +106,10 @@ class CeleryConfig(object):
 WEBDRIVER_BASEURL = "http://superset:8088/"
 # The base URL for the email report hyperlinks.
 WEBDRIVER_BASEURL_USER_FRIENDLY = WEBDRIVER_BASEURL
+# customize the polling time of each engine. The default time is 5 seconds
+DB_POLL_INTERVAL_SECONDS = {
+    "hive": int(timedelta(seconds=5).total_seconds()),
+}
```
Comment (villebro) [inline docker/pythonpath_dev/superset_config.py:112]: I'm not sure if this is needed, as it's not changing the defaults.

File: superset/db_engine_specs/hive.py [hunk 3]
```diff
@@ -375,7 +375,15 @@ def handle_cursor(  # pylint: disable=too-many-locals
                     last_log_line = len(log_lines)
                 if needs_commit:
                     session.commit()
-            time.sleep(current_app.config["HIVE_POLL_INTERVAL"])
+            if sleep_interval := current_app.config.get("HIVE_POLL_INTERVAL"):
+                logger.warning(
+                    "HIVE_POLL_INTERVAL is deprecated and will be removed in 3.0. Please use DB_POLL_INTERVAL instead"
```
Comment (villebro) [inline superset/db_engine_specs/hive.py:380]: The deprecation warning is incorrect:
```suggestion
                    "HIVE_POLL_INTERVAL is deprecated and will be removed in 3.0. Please use DB_POLL_INTERVAL_SECONDS instead"
```

File: superset/db_engine_specs/impala.py [hunk 4]
```diff
@@ -14,20 +14,31 @@
...
+logger = logging.getLogger(__name__)
+
 
 class ImpalaEngineSpec(BaseEngineSpec):
     """Engine spec for Cloudera's Impala"""
 
     engine = "impala"
     engine_name = "Apache Impala"
+    # Query 5543ffdf692b7d02:f78a944000000000: 3% Complete (17 out of 547)
+    query_progress_r = re.compile(r".*Query.*: (?P<query_progress>[0-9]+)%.*")
```
Comment (villebro) [inline superset/db_engine_specs/impala.py:41]: nit: do we really need the leading and trailing `.*` here? 
```suggestion
    query_progress_r = re.compile(r"Query.*: (?P<query_progress>[0-9]+)%")
```

Also, I know this is in line with what's being done in `hive.py`, but I would consider moving this outside the class into a constant `QUERY_PROGRESS_REGEX` in `impala.py`, as it's not defined in `BaseEngineSpec (defining it here makes it appear like we're overriding a class attribute in `BaseEngineSpec`). While at it, maybe do the same for `stage_progress_r` in `HiveEngineSpec`.

File: superset/config.py (hunk 1 above)
Comment (wanghong1314) [reply superset/config.py:1135]: OK, thank you for your suggestion. I can let the user set it without setting the default value. I want to change it to this?
DB_POLL_INTERVAL_SECONDS = {}

File: docker/pythonpath_dev/superset_config.py (hunk 2 above)
Comment (wanghong1314) [reply docker/pythonpath_dev/superset_config.py:112]: I can restore the previous code without changing this file

File: superset/config.py (hunk 1 above)
Comment (villebro) [reply superset/config.py:1135]: I would add the type as that's the convention in `config.py` and mypy can't infer the type from an empty dict:
```python
DB_POLL_INTERVAL_SECONDS: Dict[str, int] = {}
```

File: superset/db_engine_specs/hive.py (hunk 3 above)
Comment (wanghong1314) [reply superset/db_engine_specs/hive.py:380]: Sorry, I will check it carefully before submitting next time, and I will change it

File: superset/config.py (hunk 1 above)
Comment (wanghong1314) [reply superset/config.py:1135]: I changed it to this way. It's true that mypy failed

**Review by villebro — APPROVED** [review]
LGTM, thanks for all the iterations 👍

**Pull request merged**